src/ralph/virtual/models.py

```diff
@@ -0,0 +1,104 @@
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.assets.models.base import BaseObject
+from ralph.assets.models.choices import ComponentType
+from ralph.assets.models.components import GenericComponent
+from ralph.data_center.models.networks import IPAddress
+from ralph.lib.mixins.models import NamedMixin
+
+
+class CloudProvider(NamedMixin):
+    class Meta:
+        verbose_name = _('Cloud provider')
+        verbose_name_plural = _('Cloud providers')
+
+
+class CloudFlavor(BaseObject):
+    name = models.CharField(_('name'), max_length=255, unique=True)
+    cloudprovider = models.ForeignKey(CloudProvider)
+    flavor_id = models.CharField(unique=True, max_length=100)
+
+    def __str__(self):
+        return format(self.name)
+
+    @property
+    def cores(self):
+        """Number of cores"""
+        try:
+            for component in GenericComponent.objects.filter(asset=self):
+                if component.model.type == ComponentType.processor:
+                    return component.model.cores
+            return None
+        except ObjectDoesNotExist:
+            return None
+
+    @property
+    def memory(self):
+        """RAM memory size in MiB"""
+        try:
+            for component in GenericComponent.objects.filter(asset=self):
+                if component.model.type == ComponentType.memory:
+                    return component.model.size
+            return None
+        except ObjectDoesNotExist:
+            return None
+
+    @property
+    def disk(self):
+        """Disk size in MiB"""
+        try:
+            for component in GenericComponent.objects.filter(asset=self):
+                if component.model.type == ComponentType.disk:
+                    return component.model.size
+            return None
+        except ObjectDoesNotExist:
+            return None
+
+
+class CloudProject(BaseObject):
+    cloudprovider = models.ForeignKey(CloudProvider)
+    project_id = models.CharField(unique=True, max_length=100)
+    name = models.CharField(max_length=100)
+
+    def __str__(self):
+        return 'Cloud Project: {}'.format(self.name)
+
+    @property
+    def cloudhost_set(self):
+        return CloudHost.objects.filter(parent=self)
+
+
+class CloudHost(BaseObject):
+    host_id = models.CharField(unique=True, max_length=100)
+    hostname = models.CharField(max_length=100)
+    cloudflavor = models.ForeignKey(CloudFlavor, verbose_name='Instance Type')
+
+    class Meta:
+        verbose_name = _('Cloud host')
+        verbose_name_plural = _('Cloud hosts')
+
+    def __str__(self):
+        return 'Cloud Host: {}'.format(self.hostname)
+
+    @property
+    def ip_addresses(self):
+        ips = []
+        for ip in IPAddress.objects.filter(base_object=self):
+            ips.append(ip.address)
+        return ', '.join(ips)
+
+    @property
+    def service_env(self):
+        return self.parent.service_env
+
+
+class VirtualServer(BaseObject):
```

Comment:
please add some todo here that this model needs some development (ex. hostname field etc)